### PR TITLE
added 2 wait  methods

### PR DESCRIPTION
**Pull request by TyshchenkoRoma**
Waiting for Dimas approve

### CODE REVIEW

File: src/test/java/rv/atqc/gttesting/archex/AbstractPage.java [hunk 1]
```diff
@@ -1,45 +1,56 @@
...
-	}
-	
-	public T timeout(long time, TimeUnit timeUnit){
-		driver.manage().timeouts().implicitlyWait(time, timeUnit);
-		return (T) this;
-	}
-	
+    protected WebDriver driver;
+    protected JavascriptExecutor jse;
+    protected WebDriverWait wait;
```
Comment (dmatsev) [inline src/test/java/rv/atqc/gttesting/archex/AbstractPage.java]: Why you need it as field of class?

File: src/test/java/rv/atqc/gttesting/archex/AbstractPage.java [hunk 2]
```diff
@@ -1,45 +1,56 @@
...
+
+    protected AbstractPage(WebDriver driver) {
+        this.driver = driver;
+        this.wait = new WebDriverWait(driver, 10);
```
Comment (dmatsev) [inline src/test/java/rv/atqc/gttesting/archex/AbstractPage.java]: the timeout should be configurable

File: src/test/java/rv/atqc/gttesting/archex/AbstractPage.java [hunk 3]
```diff
@@ -1,45 +1,56 @@
...
+        jse.executeScript(script);
+        return (T) this;
+    }
+
+    public T act(Acting acting){
+        acting.act(driver);
+        return (T) this;
+    }
+
+    public T waitForVisibilityOfElement(WebElement webEl) {
```
Comment (dmatsev) [inline src/test/java/rv/atqc/gttesting/archex/AbstractPage.java]: please give meaningful name for webEl param

File: src/test/java/rv/atqc/gttesting/archex/AbstractPage.java [hunk 4]
```diff
@@ -1,45 +1,56 @@
...
+        wait.until(ExpectedConditions.visibilityOf(webEl));
```
Comment (dmatsev) [inline src/test/java/rv/atqc/gttesting/archex/AbstractPage.java]: I would prefer to use local wait with timeout parameter

File: src/test/java/rv/atqc/gttesting/archex/AbstractPage.java [hunk 5]
```diff
@@ -1,45 +1,56 @@
...
+        return (T) this;
+    }
+
+    public T act(Acting acting){
+        acting.act(driver);
+        return (T) this;
+    }
+
+    public T waitForVisibilityOfElement(WebElement webElement, int timeout) {
+        WebDriverWait wait =  new WebDriverWait(driver, timeout);
```
Comment (dmatsev) [inline src/test/java/rv/atqc/gttesting/archex/AbstractPage.java]: remove useless space

File: src/test/java/rv/atqc/gttesting/archex/AbstractPage.java [hunk 6]
```diff
@@ -1,45 +1,56 @@
...
+        wait.until(ExpectedConditions.visibilityOf(webElement));
+        return (T) this;
+    }
+    public T waitForElementToBeClickable(WebElement webElement, int timeout)    {
+        WebDriverWait wait =  new WebDriverWait(driver, timeout);
```
Comment (dmatsev) [inline src/test/java/rv/atqc/gttesting/archex/AbstractPage.java]: remove useless space

File: src/test/java/rv/atqc/gttesting/archex/AbstractPage.java [hunk 7]
```diff
@@ -1,45 +1,56 @@
...
+        acting.act(driver);
+        return (T) this;
+    }
+
+    public T waitForVisibilityOfElement(WebElement webElement, int timeout) {
+        WebDriverWait wait =  new WebDriverWait(driver, timeout);
+        wait.until(ExpectedConditions.visibilityOf(webElement));
+        return (T) this;
+    }
+    public T waitForElementToBeClickable(WebElement webElement, int timeout)    {
```
Comment (dmatsev) [inline src/test/java/rv/atqc/gttesting/archex/AbstractPage.java:47]: Why not timeout=10?

Comment (TyshchenkoRoma) [reply src/test/java/rv/atqc/gttesting/archex/AbstractPage.java:47]: not work 
red underline

**Review by dmatsev — APPROVED** [review]

**Pull request merged**